core/src/main/java/com/datastax/oss/driver/api/core/Version.java

```diff
@@ -23,24 +23,41 @@
 import java.util.regex.Pattern;
 
 /**
- * The version of a Cassandra release.
+ * A structured version number.
  *
- * <p>It is in the form X.Y.Z, with optional pre-release labels and build metadata.
+ * <p>Versions are expected to follow the general guidelines of<a
+ * href="https://semver.org/">semantic versioning</a>, but with a few relaxed rules.
+ *
+ * <p>The general form a version number is expected to be {@code X.Y[.Z]}, where X stands for the
+ * version's major number, Y for the minor number, and Z for an (optional) patch number.
+ *
+ * <p>It can also contain a few optional extensions: a 4th "DSE patch" number (specific to DSE
+ * releases); one or many pre-release labels; and one optional build label. See {@link
+ * #parse(String)} for a few examples.
+ *
+ * <p>This class can be used to parse versions of Apache Cassandra®, DataStax Enterprise (DSE), or
+ * the DataStax Java driver.
  *
  * <p>Version numbers compare the usual way, the major number (X) is compared first, then the minor
  * one (Y) and then the patch level one (Z). Lastly, versions with pre-release sorts before the
  * versions that don't have one, and labels are sorted alphabetically if necessary. Build metadata
- * are ignored for sorting versions.
+ * are ignored when sorting versions.
  */
-public class CassandraVersion implements Comparable<CassandraVersion> {
+public class Version implements Comparable<Version> {
 
   private static final String VERSION_REGEXP =
-      "(\\d+)\\.(\\d+)(\\.\\d+)?(\\.\\d+)?([~\\-]\\w[.\\w]*(?:\\-\\w[.\\w]*)*)?(\\+[.\\w]+)?";
+      "(\\d+)\\.(\\d+)(\\.\\d+)?(\\.\\d+)?([~\\-]\\w[.\\w]*(?:-\\w[.\\w]*)*)?(\\+[.\\w]+)?";
```

Comment:
I think I just blew a few neurons trying to parse that change 😉 

If I'm following correctly, the dash separating pre-release labels &mdash; as in "rc1-SNAPSHOT" &mdash; does not need to be escaped. Is it just unnecessary or was it failing?

Reply:
Just unnecessary (according to IntelliJ).